eras/conway/impl/testlib/Test/Cardano/Ledger/Conway/Imp/GovSpec.hs

```diff
@@ -44,20 +54,823 @@ import Test.Cardano.Ledger.Imp.Common
+          passEpoch
+          forest'' <- getProposals
+          forest'' `shouldBe` def
+        it "Subtrees are pruned when proposals expire" $ do
+          modifyPParams $ ppGovActionLifetimeL .~ EpochInterval 4
+          p1 <- submitInitConstitutionGovAction
+          passNEpochs 3
+          N
+            p11
+            [ N p111 []
+              , N p112 []
+              ] <-
+            submitConstitutionGovActionTree
+              (SJust p1)
+              $ E
+                [ E []
+                , E []
+                ]
+          N p2 [N p21 []] <-
+            submitConstitutionGovActionTree
+              SNothing
+              $ E
+                [ E []
+                ]
+          [ I []
+            , I []
+            , I []
+            , I
+                [ J
+                    p1'
+                    [ J
+                        p11'
+                        [ J p111' []
+                          , J p112' []
+                          ]
+                      ]
+                  , J p2' [J p21' []]
+                  ]
+            ] <-
+            getProposalsForest
+          SJust p1 `shouldBe` p1'
+          SJust p2 `shouldBe` p2'
+          SJust p11 `shouldBe` p11'
+          SJust p21 `shouldBe` p21'
+          SJust p111 `shouldBe` p111'
+          SJust p112 `shouldBe` p112'
+          passNEpochs 3
+          [ I []
+            , I []
+            , I []
+            , I
+                [ J p2'' [J p21'' []]
+                  ]
+            ] <-
+            getProposalsForest
+          SJust p2 `shouldBe` p2''
+          SJust p21 `shouldBe` p21''
+        it "Subtrees are pruned when competing proposals are enacted" $ do
+          (dRep, committeeMember) <- electBasicCommittee
+          [ N
+              p1
+              [ N
+                  p11
+                  [ N p111 []
+                    , N p112 []
+                    ]
+                ]
+            , N p2 [N p21 []]
+            ] <-
+            submitConstitutionGovActionForest
+              SNothing
+              [ E
+                  [ E
+                      [ E []
+                      , E []
+                      ]
+                  ]
+              , E
+                  [ E []
+                  ]
+              ]
+          [ I []
+            , I []
+            , J _ [] -- Due to the committee enactment
+            , I
+                [ J
+                    p1'
+                    [ J
+                        p11'
+                        [ J p111' []
+                          , J p112' []
+                          ]
+                      ]
+                  , J p2' [J p21' []]
+                  ]
+            ] <-
+            getProposalsForest
+          SJust p1 `shouldBe` p1'
+          SJust p2 `shouldBe` p2'
+          SJust p11 `shouldBe` p11'
+          SJust p21 `shouldBe` p21'
+          SJust p111 `shouldBe` p111'
+          SJust p112 `shouldBe` p112'
```

Comment:
```suggestion
          a <-
            submitConstitutionGovActionForest
              SNothing
              [ E
                  [ E
                      [ E []
                      , E []
                      ]
                  ]
              , E
                  [ E []
                  ]
              ]
          b <-
            getProposalsForest
          fmap SJust a `shouldBe` b
```

Would this work?

Reply:
Awesome! Thank you! Yes, this would work I will do this in as many places as is possible. We also discuss your point during our 1 on 1, Alexey and I :+1: